app/domain/rotation/conjur_facade.rb

```diff
@@ -38,15 +44,24 @@ def current_values(variables)
     #
     def update_variables(new_values, &rotator_code)
       @db.transaction do
+
+        # lock_secrets_for_resource_ids(new_values.keys)
+
         new_values.each do |resource_id, value|
           update_secret(resource_id, value)
         end
         rotator_code.call if rotator_code
       end
+    rescue => e
+      puts "Rotator update failed: #{e}"
```

Comment:
How can we ever wind up here? If there's no way, this `rescue` should get removed. If there's some way, it needs to do more error handling than just a `puts`.

Reply:
We can definitely wind up here.  When Rafal integrates the audit code, he'll audit this event.

Reply:
It needs to do more than audit, though, doesn't it? If rotation fails, it needs to reset the expiration.

Reply:
Yep, you're correct.  I forgot I added that during debugging.  I've just removed it and repushed, so now the errors will be caught by this part in MasterRotator, as intended:
```ruby
      def run
        rotator.rotate(@facade)
      rescue => e
        p "Catching error", e
        set_retry_expiration
        log_error(e)
      end
```